pkg/cmd/issue/shared/lookup.go

```diff
@@ -12,49 +12,48 @@ import (
 )
 
 func IssueFromArg(apiClient *api.Client, baseRepoFn func() (ghrepo.Interface, error), arg string) (*api.Issue, ghrepo.Interface, error) {
-	issue, baseRepo, err := issueFromURL(apiClient, arg)
-	if err != nil {
-		return nil, nil, err
-	}
-	if issue != nil {
-		return issue, baseRepo, nil
-	}
-
-	baseRepo, err = baseRepoFn()
-	if err != nil {
-		return nil, nil, fmt.Errorf("could not determine base repo: %w", err)
+	issueNumber, baseRepo := issueMetadataFromURL(arg)
+
+	if baseRepo == nil {
+		var err error
+		baseRepo, err = baseRepoFn()
+		if err != nil {
+			return nil, nil, fmt.Errorf("could not determine base repo: %w", err)
+		}
 	}
 
-	issueNumber, err := strconv.Atoi(strings.TrimPrefix(arg, "#"))
-	if err != nil {
-		return nil, nil, fmt.Errorf("invalid issue format: %q", arg)
+	if issueNumber == 0 {
```

Comment:
Could this check happen before the `if baseRepo == nil` check?

It's an edge case, but if both issue number is malformed and the base repo could not be detected (for whatever reason), we would want the error about the issue number format to be raised instead of the one about the base repo.